client/src/main/java/com/vesoft/nebula/client/graph/PooledGraphClient.java

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package com.vesoft.nebula.client.graph;
+
+import com.facebook.thrift.TException;
+import com.vesoft.nebula.graph.ErrorCode;
+import com.vesoft.nebula.utils.GraphConstants;
+import java.util.Properties;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Graph Client with thread safe and connection failover able
+ */
+public class PooledGraphClient implements GraphClient {
```

Comment:
I don't know why the connection pool will implement Client

Reply:
In order to unify the upper interface

Reply:
It's unnecessary... we will using connection pool to get a client from it.